charm/kafka/lib/charms/layer/kafka.py

```diff
@@ -90,18 +107,70 @@ def configure_kafka(self, zk_units, log_dir=None):
         self.restart()
 
     def restart(self):
-        self.stop()
-        self.start()
+        '''
+        Restarts the Kafka service.
+        '''
+        host.service_restart(KAFKA_SERVICE)
 
     def start(self):
-        host.service_start(KAFKA_SERVICE)
+        '''
+        Starts the Kafka service.
+        '''
+        host.service_reload(KAFKA_SERVICE)
 
     def stop(self):
+        '''
+        Stops the Kafka service.
+
+        '''
         host.service_stop(KAFKA_SERVICE)
-    
+
     def is_running(self):
+        '''
+        Restarts the Kafka service.
+        '''
         return host.service_running(KAFKA_SERVICE)
 
+    def get_zks(self):
```

Comment:
Can remove this. Zookeeper relation data is always passed in to Kafka.install. Don't need to make it like the Zookeeper charm -- where this is a peer relation with different semantics & behavior. Existing behavior is known to work; if it's not broken, no need to change it.

Reply:
Nevermind.. this supports cases where we want to get the zookeeper relation data outside of a @when on a flag with that relation data. This is a good fallback.